websites/H/HentaiVN/presence.ts

```diff
@@ -0,0 +1,578 @@
+const presence = new Presence({
+		clientId: "942782020055089192"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let namekey, doukey, charkey, keystr, largebox;
+
+presence.on("UpdateData", async () => {
+	const { pathname, search } = document.location,
+		searchParams = new URLSearchParams(search),
+		page = searchParams.get("page") || "1",
+		presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		};
+
+	if (pathname.startsWith("/tim-kiem-truyen.html") && searchParams.has("key")) {
+		presenceData.details = "Đang tìm kiếm từ khóa:";
+		presenceData.state = `${
+			document.querySelector("input").value
+		} - Trang ${page}`;
+		presenceData.smallImageKey = "search";
+		presenceData.largeImageKey = "home";
+	} else if (
+		pathname.startsWith("/tim-kiem-tac-gia.html") &&
+		searchParams.has("key")
+	) {
+		presenceData.details = "Đang tìm kiếm truyện của tác giả:";
+		presenceData.state = `${
+			document.querySelector("input").value
+		} - Trang ${page}`;
+		presenceData.smallImageKey = "search";
+		presenceData.largeImageKey = "home";
+	} else if (
+		pathname.startsWith("/tim-kiem-doujinshi.html") &&
+		searchParams.has("key")
+	) {
+		presenceData.details = "Đang tìm kiếm truyện là Doujin của:";
+		presenceData.state = `${
+			document.querySelector("input").value
+		} - Trang ${page}`;
+		presenceData.smallImageKey = "search";
+		presenceData.largeImageKey = "home";
+	} else if (
+		pathname.startsWith("/tim-kiem-charater.html") &&
+		searchParams.has("key")
+	) {
+		presenceData.details = "Đang tìm kiếm truyện có nhân vật:";
+		presenceData.state = `${searchParams.get("key")} - Trang ${page}`;
+		presenceData.smallImageKey = "search";
+		presenceData.largeImageKey = "home";
+	} else if (pathname === "/") {
+		presenceData.details = "Đang duyệt trang chủ";
+		presenceData.state = `Trang ${page}`;
+		presenceData.smallImageKey = "surf";
+		presenceData.largeImageKey = "home";
+	} else if (pathname.startsWith("/forum/quote.php")) {
+		presenceData.details = "Đang trả lời bình luận";
+		presenceData.state = `Bình luận${
+			document
+				.querySelector("div[class='box-title']")
+				.textContent.trim()
+				.split("FORUM\n/\nQuay về topic\n/\nTrả lời kèm trích dẫn")[1]
+		}`;
+		presenceData.smallImageKey = "bell";
+		presenceData.largeImageKey = "forum";
+	} else if (document.location.search.includes("id")) {
```

Comment:
```suggestion
	} else if (searchParams.has("id")) {
```
Some suggestion, should work the same & eliminate false positives

Reply:
If not, replace `document.location.search` with `search`, which is the variable you declared

Reply:
thanks